alvr/system_info/src/android.rs

```diff
@@ -1,9 +1,45 @@
 use alvr_common::warn;
-use jni::{objects::JObject, sys::jobject, JNIEnv, JavaVM};
+use jni::{
+    errors::Error,
+    objects::{JObject, JString, JValue, JValueOwned},
+    sys::jobject,
+    JNIEnv, JavaVM,
+};
 use std::net::{IpAddr, Ipv4Addr};
 
 pub const MICROPHONE_PERMISSION: &str = "android.permission.RECORD_AUDIO";
 
+// Will call a JNI method and handle any Java exceptions that might occur
+fn env_call_method_handled<'local>(
+    env: &mut JNIEnv<'local>,
+    object: &JObject,
+    method: &str,
+    sig: &str,
+    args: &[JValue],
+) -> Result<JValueOwned<'local>, Error> {
```

Comment:
Usually i follow the path of fewer parameters when making a function to abstract away some code. In this case it would be much easier to make a function taking a `jni::Result<T>` and return a `anyhow::Result<T>`. Inside the function, don't panic because you don't know the intention of the caller, even if we always unwrap, don't expect anything. Instead limit this function to just transform the error.

Actually, before doing this, check if just calling `.to_any()` at the caller site is enough (with the purpose of making the error message clearer). This function is defined in alvr_common. Other things you could do is to play with different printing methods (using `{}` or `{:?}` placeholders with `format!()`)